cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/PropertiesResolver.java

```diff
@@ -62,8 +63,13 @@
     systemArgs.put(Constants.CLUSTER_NAME, cConf.get(Constants.CLUSTER_NAME, ""));
     systemArgs.put(Constants.AppFabric.APP_SCHEDULER_QUEUE, queueResolver.getQueue(id.getNamespace()));
     if (SecurityUtil.isKerberosEnabled(cConf)) {
-      ImpersonationInfo impersonationInfo = new ImpersonationInfo(
-        namespaceQueryAdmin.get(id.toEntityId().getNamespaceId()), cConf);
+      ImpersonationInfo impersonationInfo;
```

Comment:
That would be preferable, but OwnerAdmin was not ready when I added this, I will open a JIRA and we can refactor this.